File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/SpecificationService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.SpecificationVO;
import com.offcn.pojo.TbSpecification;

import java.util.List;

public interface SpecificationService {
    public List<TbSpecification> findAll();
    public void add(SpecificationVO specificationVO);
    public SpecificationVO findOne(Long id);
    public void updateSpecificationAndOption(SpecificationVO specificationVO);

    /**
     *  删除规格以及级联删除规格选项
     * @param id 规格的主键
     */
    public void delete(Long[] id);

    /**
     * 条件查询
     * @return
     */
    public List<TbSpecification> findAll(TbSpecification specification);
}

File: youlexuan-manager-web/src/main/webapp/js/service/specificationService.js
app.service('specificationService',function ($http) {
    this.findAll = function () {
       return $http.get('../specification/findAll.do');
    }

    this.add = function (entity) {
        return $http.post('../specification/add.do',entity);
    }
    
    
    this.findOne = function (id) {
        return $http.get('../specification/findOne.do?id='+id);
    }
    
    this.update = function (entity) {
        return $http.post('../specification/update.do',entity);
    }

    this.delete = function (ids) {
        return $http.get('../specification/delete.do?ids='+ids);
    }
    
    this.findByCondition = function (searchEntity) {
        return $http.post('../specification/findByCondition.do',searchEntity);
    }
})